dali/operators/reader/nvdecoder/cuvideodecoder.cc

```diff
@@ -161,12 +161,16 @@ void CUVideoDecoder::reconfigure(unsigned int height, unsigned int width) {
 }
 
 int CUVideoDecoder::initialize(CUVIDEOFORMAT* format) {
-    if (decoder_) {
+    while (decoder_) {
```

Comment:
Why `while` if there is `return` at the end of this loop?

Reply:
Oh, I missed the break. IMO, the logic is a bit convoluted, this is a bit `goto` like :P 
Can we somehow make it into
```
if (some condition)
   FAIL
if (other condition) 
   do something;
   return 1;
if (third condition)
   clear the state & continue with config?
```